test/unit/init/CliRunner.test.ts

```diff
@@ -182,4 +182,34 @@ describe('CliRunner', (): void => {
     expect(exit).toHaveBeenCalledTimes(1);
     expect(exit).toHaveBeenCalledWith(1);
   });
+
+  it('exits when unknown parameters are passed to the main executable.', async(): Promise<void> => {
+    new CliRunner().run({
+      argv: [
+        'node', 'script', '--foo',
+      ],
+    });
+
+    // Wait until initializer has been called, because we can't await CliRunner.run.
```

Comment:
Maybe we should just change that, actually?

Reply:
@arnevandoorslaer I think it should be easy to also add this change in this PR. Change `public run` in CliRunner to `public async run` and have it await/return the `this.createInitializer` call. Might perhaps also have to change a few tests to then await the run call in this file.

Reply:
Btw, let me know if I'm wrong and this is more complicated than I thought. No reason to waste too much time on this.

Reply:
FYI, I deliberately made `run` to be non-async as to lower the chance of unhandled promise rejection errors in the future.
While there is no problem with making this async atm (`createInitializer` has a `.catch`), there is a danger that another `await` may be added in `run` in the future without an explicit `.catch`. In that case, an unhandled promise rejection error may be thrown.

Could also be fixed with a comment of course, but having the compiler enforce it is always safer :-)

Reply:
After #652 I agree that it's better to make it sync again and add a comment why it is sync.